koiki/create_delivery.py

```diff
@@ -1,29 +1,61 @@
-from koiki.sender import Sender
-from koiki.recipient import Recipient
-from koiki.order import Order
+from koiki.models import Sender, Recipient, Shipment
+from koiki.woocommerce.models import LineItem, Shipping, Billing
 
 
 class CreateDelivery():
     LABEL_FORMAT = 'PDF'
     RESOURCE_PATH = '/altaEnvios'
 
     def __init__(self, order):
-        self.order = Order(order)
-        self.recipient = Recipient(order)
-        self.sender = Sender()
+        self.order = order
+        self.by_vendor = self._by_vendor(order['line_items'])
+        self.shipping = Shipping(order['shipping'])
+        self.billing = Billing(order['billing'])
 
     def body(self):
         return {
             'formatoEtiqueta': self.LABEL_FORMAT,
-            'envios': [self._delivery()]
+            'envios': self._deliveries()
         }
 
     def url(self):
         return self.RESOURCE_PATH
 
-    def _delivery(self):
+    # Builds a single delivery for each vendor, aggregating the line items that vendor sold
+    def _deliveries(self):
+        deliveries = []
+
+        for vendor_id in self.by_vendor.keys():
+            line_items = self.by_vendor[vendor_id]
+            vendor = line_items[0].vendor
+
+            deliveries.append(self._delivery(line_items, vendor))
+
+        return deliveries
+
+    def _by_vendor(self, line_items):
+        by_vendor = {}
+
+        for line_item in line_items:
+            item = LineItem(line_item)
+            vendor_id = item.vendor.id
+
+            if vendor_id in by_vendor.keys():
```

Comment:
This operations are normally quite common, I prefer to use the https://www.geeksforgeeks.org/defaultdict-in-python/  way to simplify the ifs in the logic.

from collections import defaultdict
by_vendor = defaultdict(list)
.....
for line_item in line_items:
            item = LineItem(line_item)
            vendor_id = item.vendor.id
            by_vendor[vendor_id].append(item)

Reply:
Nice! I did work! https://github.com/coopdevs/lazona_connector/pull/45/commits/6ffbad6030bea69bd249988010e16adb4a0c4abc